### PR TITLE
updated installation documentation link in readme

### LINKED ISSUE #1176: Update README.md link to _installation documentation_.
[README.md](https://github.com/zarr-developers/zarr-python#readme) installation documentation points to https://zarr.readthedocs.io/en/stable/#installation. 
I was wondering if it needed to point to the updated documentation at https://zarr.readthedocs.io/en/latest/installation.html.

Also the other links on the [README.md](https://github.com/zarr-developers/zarr-python#readme) page point to the _stable_branch_ link on _readthedocs_.
I was thinking it would be good to have them point to _latest_branch_ rather

### PR DESCRIPTION
**Pull request by DON-BRAN**
fixes #1176

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [ ] GitHub Actions have all passed
* [ ] Test coverage is 100% (Codecov passes)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -107,4 +107,4 @@ or via `conda`:
 conda install -c conda-forge zarr
 ```
 
-For more details, including how to install from source, see the [installation documentation](https://zarr.readthedocs.io/en/stable/#installation).
+For more details, including how to install from source, see the [installation documentation](https://zarr.readthedocs.io/en/latest/installation.html).
```
Comment (jakirkham) [inline README.md:110]: Think we want to keep this as `stable` per ( https://github.com/zarr-developers/zarr-python/issues/1176#issuecomment-1276572422 )

```suggestion
For more details, including how to install from source, see the [installation documentation](https://zarr.readthedocs.io/en/stable/installation.html).
```

cc @joshmoore

Comment (joshmoore) [reply README.md:110]: Definitely. :+1:

File: README.md [hunk 2]
```diff
@@ -107,4 +107,4 @@ or via `conda`:
 conda install -c conda-forge zarr
 ```
 
-For more details, including how to install from source, see the [installation documentation](https://zarr.readthedocs.io/en/stable/#installation).
+For more details, including how to install from source, see the [installation documentation](https://zarr.readthedocs.io/en/stable/installation.html).
```
Comment (jakirkham) [inline README.md:110]: Tried this link and it seems to not work for me. This appears to though. Sorry for not catching this earlier.

```suggestion
For more details, including how to install from source, see the [installation documentation](https://zarr.readthedocs.io/en/stable/index.html#installation).
```

Comment (DON-BRAN) [reply README.md:110]: You're right. 
I think this update will be best when the changes in 'latest' have been brought into "stable"
@jakirkham will it be alright if I close the PR and reopen when changes in 'latest' have been brought into "stable" or i keep it open?

**Review by jakirkham — APPROVED** [review]

**Pull request merged**